me.id.webverify/webverifylib/src/main/java/me/id/webverifylib/IDmeProfile.java

```diff
@@ -1,5 +1,138 @@
 package me.id.webverifylib;
 
-public class IDmeProfile {
-  // TODO mirland 13/12/16:
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.NonNull;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Locale;
+
+public final class IDmeProfile implements Parcelable {
+  private String id;
+  private boolean verified;
+  private String affiliation;
+  private String fname;
+  private String lname;
+  private String zip;
+  private String email;
+  private String uuid;
+  private String group;
+
+  IDmeProfile(@NonNull String json) throws JSONException {
+    JSONObject jsonProfile = new JSONObject(json);
+    id = getJsonField(jsonProfile, "id");
+    verified = jsonProfile.getBoolean("verified");
+    affiliation = getJsonField(jsonProfile, "affiliation");
+    fname = getJsonField(jsonProfile, "fname");
+    lname = getJsonField(jsonProfile, "lname");
+    zip = getJsonField(jsonProfile, "zip");
+    email = getJsonField(jsonProfile, "email");
+    uuid = getJsonField(jsonProfile, "uuid");
+    group = getJsonField(jsonProfile, "group");
+  }
+
+  private String getJsonField(JSONObject jsonProfile, String id) {
+    String field = null;
+    try {
+      field = jsonProfile.getString(id);
+    } catch (JSONException e) {
+      e.printStackTrace();
+    }
+    return field == null || field.equals("null") ? null : field;
```

Comment:
This doesn't look so good to me, we should avoid this kind of stuff on mobile client right? Maybe is something that the BE should fix. Is not so common I guess, but what if your name is null? :)